app/scripts/models/Workspace.js

```diff
@@ -98,6 +98,22 @@ define(['backbone', 'Nodes', 'Connection', 'Connections', 'scheme', 'FLOOD', 'Ru
         this.listenTo(this, 'change:workspaceDependencyIds', throttledSync);
         this.listenTo(this, 'requestRun', this.run);
 
+        this.listenTo(this.get('nodes'), 'change:selected', function (e) {
```

Comment:
App does not know about nodes but it knows about workspaces, so traverse events from node to app through workspace.